amethyst_animation/src/lib.rs

```diff
@@ -45,8 +45,9 @@
 //! [ex_ani]: https://github.com/amethyst/amethyst/tree/master/examples/animation
 //! [ex_gltf]: https://github.com/amethyst/amethyst/tree/master/examples/gltf
 
-#![warn(missing_docs)]
-#![cfg_attr(feature = "cargo-clippy", allow(type_complexity))] // complex project
+#![warn(missing_docs, rust_2018_idioms, rust_2018_compatibility)]
+// I have to comment this out as right now no version of this doesn't generate warnings on both stable and beta.
+// #![allow(clippy::type_complexity)] // complex project
```

Comment:
I would say we should remove this completely for now and properly integrate clippy in another PR.

Reply:
Leaving the comment there is valuable as it serves as a reminder to uncomment it later imo.

Reply:
I think that stuff should go in an issue and not litter the code personally.

Reply:
Ok fair I'll remove it.